tests/console/suseconnect_scc.pm

```diff
@@ -31,17 +30,25 @@ sub run {
 
     select_serial_terminal;
     die 'SUSEConnect package is not pre-installed!' if script_run 'command -v SUSEConnect';
-    if ((is_jeos || is_sle_micro) && script_run(qq($reg_cmd --status-text | grep -i 'not registered'))) {
+    if ((is_jeos || is_sle_micro) && script_run(qq(SUSEConnect --status-text | grep -i 'not registered'))) {
         die 'System has been already registered!';
     }
 
     # There are sporadic failures due to the command timing out, so we increase the timeout
     # and make use of retries to overcome a possible sporadic network issue.
-    my $output = script_output_retry("$cmd", retry => $retries, delay => $delay, timeout => 180);
-    die($output) if ($output =~ m/error|timeout|problem retrieving/i);
-    process_reboot(trigger => 1) if is_transactional;
+    # script_output_retry is useless for `transactional-update` cmd because it returns 0 even with failure
+    # trup_call will raise a failure if the command fails
+    if (is_transactional) {
+        trup_call('register' . $cmd);
+        trup_call('--continue run zypper --gpg-auto-import-keys refresh');
```

Comment:
Only do `--gpg-auto-import-keys` if *really* necessary

Reply:
you mean with something like `is_staging` ?

Reply:
Yes. Arguably it should also verify that it's the correct key somehow, but not sure how to get that info

Reply:
done